src/annotation.js

```diff
@@ -142,8 +142,10 @@ function draw(chart, caller, clip) {
     clipArea(ctx, chartArea);
   }
 
-  drawElements(ctx, visibleElements, caller);
-  drawSubElements(ctx, visibleElements, caller);
+  const todraw = drawElements(ctx, visibleElements, caller);
+  for (const item of todraw) {
+    item.element.draw(ctx, item.box);
+  }
```

Comment:
@kurkle I have changed the cycle to draw the elements in order to have an array with all elements (ans sub ones) to be drawn. This will enable also the z option in next PR. I can come back to previous one if you prefer

Reply:
Please try to separate things. It might be useful for z-index to create an array, but it has no use whatsoever in this PR.

Reply:
ok, apologize

Reply:
fall back to previous commit